app/src/main/java/com/github/se/cyrcle/ui/theme/molecules/Filter.kt

```diff
@@ -263,3 +323,142 @@ fun FilterSection(
+                          showSuggestions.value = false
+                        }
+                        .testTag("ClearSearchButtonListScreen"))
+          }
+        },
+        singleLine = true,
+        keyboardOptions = KeyboardOptions.Default.copy(imeAction = ImeAction.Search),
+        keyboardActions =
+            KeyboardActions(
+                onSearch = {
+                  virtualKeyboardManager?.hide()
+
+                  runBlocking { addressViewModel.search(textFieldValue.value) }
+
+                  showSuggestions.value = true
+                }))
+
+    if (listOfSuggestions.value.size + 1 > 0) {
+      DropdownMenu(
+          expanded = showSuggestions.value,
+          onDismissRequest = { showSuggestions.value = false },
+          modifier =
+              Modifier.width(with(LocalDensity.current) { textFieldSize.value.width.toDp() })
+                  .testTag("SuggestionsMenuListScreen")) {
+            if (permissionHandler.getLocalisationPerm().collectAsState().value) {
+              DropdownMenuItem(
+                  text = { Text("My Location") },
+                  modifier = Modifier.testTag("MyLocationSuggestionMenuItem"),
+                  onClick = {
+                    myLocation.value = true
+                    showSuggestions.value = false
+                    textFieldValue.value = "My Location"
+                    isTextFieldVisible.value = false
+                    focusManager.clearFocus()
+                  })
+            }
+
+            val seenNames = mutableSetOf<String>()
+            uniqueSuggestions.value =
+                listOfSuggestions.value.filter { suggestion ->
+                  val displayName =
+                      suggestion.suggestionFormatDisplayName(
+                          maxSuggestionDisplayNameLengthList, "ListScreen")
+                  if (displayName in seenNames) {
+                    false
+                  } else {
+                    seenNames.add(displayName)
+                    true
+                  }
+                }
+
+            for (address in uniqueSuggestions.value.take(NumberOfSuggestionsForMenu)) {
+              DropdownMenuItem(
+                  text = {
+                    Text(
+                        address.suggestionFormatDisplayName(
+                            maxSuggestionDisplayNameLengthList, "ListScreen"))
+                  },
+                  modifier = Modifier.testTag("SuggestionMenuItem${address.city}"),
+                  onClick = {
+                    chosenLocation.value = address
+                    showSuggestions.value = false
+                    myLocation.value = false
+                    isTextFieldVisible.value = false
+                    textFieldValue.value =
+                        address.suggestionFormatDisplayName(
+                            maxSuggestionDisplayNameLengthList, "ListScreen")
+                    focusManager.clearFocus()
+                  })
```

Comment:
Here i think for the readability of the suggestions it is very important to add `contentPadding = PaddingValues(8.dp)`

Reply:
fixed !